src/Stateless/Reflection/MethodInfo.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Stateless.Reflection
+{
+    /// <summary>
+    /// Describes an internal method through the reflection API.
+    /// </summary>
+    public class MethodInfo
```

Comment:
Unfortunately the name `MethodInfo` overlaps with `System.Reflection.MethodInfo`; the namespace does disambiguate them, but seeing `MethodInfo` in any reflection-related code would immediately suggest the framework class to me. What other contenders do we have for this name? `InvocationInfo` was one, any others?